ocs_ci/deployment/deployment.py

```diff
@@ -1172,6 +1175,94 @@ def deploy_acm_hub(self):
         """
         Handle ACM HUB deployment
         """
+        if config.ENV_DATA.get("acm_hub_downstream"):
+            self.deploy_acm_hub_downstream()
+        else:
+            self.deploy_acm_hub_upstream()
+
+    def deploy_acm_hub_downstream(self):
+        """
+        Handle AM HUB downstream deployment
+        """
+        logger.info("Cloning open-cluster-management deploy repository")
+        acm_hub_downstream_deploy_dir = os.path.join(
+            constants.EXTERNAL_DIR, "acm_hub_downstream_deploy"
+        )
+        clone_repo(
+            constants.ACM_HUB_DOWNSTREAM_DEPLOY_REPO, acm_hub_downstream_deploy_dir
+        )
+
+        logger.info("Retrieving quay token")
+        docker_config = load_auth_config().get("quay", {}).get("cli_password", {})
+        pw = base64.b64decode(docker_config)
+        pw = pw.decode().replace("quay.io", "quay.io:443").encode()
+        quay_token = base64.b64encode(pw).decode()
+
+        kubeconfig_location = os.path.join(self.cluster_path, "auth", "kubeconfig")
+
+        logger.info("Setting env vars")
+        env_vars = {
+            "QUAY_TOKEN": quay_token,
+            "COMPOSITE_BUNDLE": "true",
+            "CUSTOM_REGISTRY_REPO": "quay.io:443/acm-d",
+            "DOWNSTREAM": "true",
+            "DEBUG": "true",
+            "KUBECONFIG": kubeconfig_location,
+        }
+        for key, value in env_vars.items():
+            if value:
+                os.environ[key] = value
+
+        logger.info("Writing pull-secret")
+        _templating = templating.Templating(
+            os.path.join(constants.TEMPLATE_DIR, "acm-deployment")
+        )
+        template_data = {"docker_config": docker_config}
+        data = _templating.render_template(
+            constants.ACM_HUB_DOWNSTREAM_PULL_SECRET_TEMPLATE,
+            template_data,
+        )
+        pull_secret_path = os.path.join(
+            acm_hub_downstream_deploy_dir, "prereqs", "pull-secret.yaml"
+        )
+        with open(pull_secret_path, "w") as f:
+            f.write(data)
+
+        logger.info("Creating ImageContentSourcePolicy")
+        run_cmd(f"oc create -f {constants.ACM_HUB_DOWNSTREAM_ICSP_YAML}")
+
+        logger.info("Writing tag data to snapshot.ver")
+        # TODO: pull from config once this is set in OCP configs
+        tag_data = "2.4.3-DOWNSTREAM-2022-04-02-03-50-40"
+        with open(
+            os.path.join(acm_hub_downstream_deploy_dir, "snapshot.ver"), "w"
+        ) as f:
+            f.write(tag_data)
+
+        ci_dir = os.getcwd()
+        os.chdir(acm_hub_downstream_deploy_dir)
+        logger.info("Running open-cluster-management deploy")
+        cmd = [os.path.join(acm_hub_downstream_deploy_dir, "start.sh"), "--silent"]
+        logger.info("Running cmd: %s", " ".join(cmd))
+        proc = Popen(
+            cmd,
+            stdout=PIPE,
+            stderr=PIPE,
+            encoding="utf-8",
```

Comment:
Not sure if it can be the case but can you try with shell=True?

```
If shell is True, the specified command will be executed through the shell. This can be useful if you are using Python primarily for the enhanced control flow it offers over most system shells and still want convenient access to other shell features such as shell pipes, filename wildcards, environment variable expansion, and expansion of ~ to a user’s home directory. However, note that Python itself offers implementations of many shell-like features (in particular, [glob](https://docs.python.org/3/library/glob.html#module-glob), [fnmatch](https://docs.python.org/3/library/fnmatch.html#module-fnmatch), [os.walk()](https://docs.python.org/3/library/os.html#os.walk), [os.path.expandvars()](https://docs.python.org/3/library/os.path.html#os.path.expandvars), [os.path.expanduser()](https://docs.python.org/3/library/os.path.html#os.path.expanduser), and [shutil](https://docs.python.org/3/library/shutil.html#module-shutil)).
```

In the script itself they are using the pipes and so on, but not sure if it might be the case if those are not directly used in the command.

Have you tried the debug output with added:

set -x 

Or you can directly run command like:

 bash -x script-name.sh
 
 this will not need you to change the script which might be better option for you.
 
 I am really curious to see debug output to see what happens inside the script.